controllers/admin_client.go

```diff
@@ -695,6 +696,10 @@ func (client *MockAdminClient) ExcludeInstances(addresses []string) error {
 	exclusionMap := make(map[string]bool, count)
 	newExclusions := make([]string, 0, count)
 	for _, address := range addresses {
+		if !isValidAddress(address) {
```

Comment:
I realise I'm new to this code base, but these mocks seem very heavy - using testify mock would give finer control and avoid duplicating business logic that exists within fdbcli. I'd be happy to whip up an example patch for you if you'd like.

Reply:
My personal preference is for more heavyweight mocks that give us a more realistic simulation of the service that we're stubbing out.